### ISSUE TITLE
Memory/dataset size limitations

### ISSUE THREAD

**Issue by mpetkova** [ashwin/gDel3D#6]
Hello, 

I'm running into memory issues when running the code on AWS instances with 8GB + of GPU memory. I am working with large datasets (they can be up to 200,000,000 points), so I tried testing gDel3D on 10,000,000 randomly-generated points. I got the following error, which was thrown when the code reached ~8GB of memory use, even on 16GB GPU instances:
    
`cudaCheckError() with sync failed at /home/ubuntu/gDel3D/GDelFlipping/src/gDel3D/GpuDelaunay.cu:901 : an illegal memory access was encountered`

Is there a hard limit to the number of points gDel3D can be run on, or are there any workarounds?

Thank you

**Comment by ghemann** [comment]
I agree, this is a problem for me as well.  Any resolutions to this?

**Comment by JagnDC** [comment]
I also encountered the same problem, is there any solution?

**Comment by JagnDC** [comment]
I'm working with cuda9.0(1080ti). When I set pointnum=17000000, It will form an endless cycle. 
Could someone solve the problem, please tell me ,thank u !

**Comment by QiuYuanHan** [comment]
Yes indeed, This project use the upper 26 bits to store tet id, so there is an inherient limitition of around 800w（7GB video mem） points. tets_num(points_num*8.5) surpass 【2^(31-5)】 would trigger an int overflow.
check CommonTyps.h
```cpp
///////////////////////////////////////////////////////////////////// TetOpp //
// ...76543210
//       ^^^^^ vi (2 bits)
//       |||__ internal
//       ||___ special
//       |____ sphere fail
// Rest is tetIdx

__forceinline__ __host__ __device__ int getOppValTet( int val )
{
    return ( val >> 5 );
}

__forceinline__ __host__ __device__ void setOppValTet( int &val, int idx )
{
    val = ( val & 0x1f ) | ( idx << 5 ); 
}
```
overflow usually happends in setOppValTet. So you could either chunk your data or change the TetOpp structure (Both turned out to be a PITA).
